server/e2e_test.go

```diff
@@ -214,6 +219,22 @@ func TestPassivePlayersGetBanEffects(t *testing.T) {
 	h.WaitEmptyInboxes(allClients)
 }
 
+// TestAbleToFindDisconnectedPlayerInFrozenSnapshots confirms that even if a
+// user disconnects after sending a message, the server will still be able to
+// find them if they were in a frozen pool.
+func TestAbleToFindDisconnectedPlayerInFrozenSnapshots(t *testing.T) {
+	h := newTestHarness(t, basicPoolSize)
+	clients := h.NewPool(basicPoolSize, testAmount, testVersion, nil)
+	disconnecter := clients[0]
+	expectedSession := disconnecter.session
+	assert.NotEmpty(t, expectedSession)
+
+	// disconnect and see if we can find the player
+	disconnecter.Disconnect()
+	found := h.tracker.findPlayerInPoolSnapshots(disconnecter.remoteConn)
+	assert.Equal(t, expectedSession, found.sessionID)
+}
+
```

Comment:
Testing the race condition of send+disconnect (described below) would require some hooks or breaking up functions. Instead of that I just test that we can actually look up disconnected players.

Reply:
I think we can move the find disconnected player to another PR. I am afraid we will get false positives.